hooks/hiero_customize_export_ui.py

```diff
@@ -0,0 +1,286 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+#  UI Hook
+# ===========================
+
+import sgtk
+
+from sgtk.platform.qt import QtGui
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class HieroCustomizeExportUI(HookBaseClass):
+    """
+    This class defines methods that can be used to customize the UI of the various
+    Shotgun-related exporters. Each processor has its own set of create/get/set
```

Comment:
the separate examples snippets are great but i think adding an full example would be really useful too. Maybe even something that could be cut n pasted into a hook file as a starting point for TDs!